collector/receiver/otelarrowreceiver/README.md

```diff
@@ -1,89 +1,108 @@
-format.
+protocol via gRPC or HTTP.
 
 ## Getting Started
 
-All that is required to enable the OTLP receiver is to include it in the
-receiver definitions. A protocol can be disabled by simply not specifying it in
-the list of protocols.
+The OTel-Arrow receiver is an extension of the core OpenTelemetry
+Collector [OTLP
+receiver](https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/otlpreceiver)
+component with additional support for the
+[OTel-Arrow](https://github.com/open-telemetry/otel-arrow) protocol.
+
+OTel-Arrow supports column-oriented data transport using the Apache
+Arrow data format.  The OTel-Arrow exporter converts OTLP data into an
+optimized representation and then sends rows of data using Apache
+Arrow to encode the stream.  This component contains logic to reverse
+the process used in the OTel-Arrow exporter.
+
+The use of an OTel-Arrow exporter-receiver pair is recommended when
+the network is expensive.  Typically, expect to see a 50% reduction in
+bandwidth compared with the same data being sent using standard OTLP,
+unary gRPC, and zstd compression.
+
+This component includes all the features and configuration of the core
+OTLP receiver, making it possible to upgrade from the core component
+simply by replacing "otlp" with "otelarrow" as the component name in
+the collector configuration.
+
+To enable the OTel-Arrow receiver, include it in the list of receivers
+for a pipeline.  Arrow functionality is enabled by default, but like the
+component it is based on, each protocol must be listed.
 
 ```yaml
 receivers:
-  otlp:
+  otelarrow:
     protocols:
       grpc:
       http:
 ```
 
-The following settings are configurable:
+The above configuration uses the OTLP-specified default ports 4317
+(gRPC) and 4318 (HTTP).
 
-- `endpoint` (default = 0.0.0.0:4317 for grpc protocol, 0.0.0.0:4318 http protocol):
-  host:port to which the receiver is going to receive data. The valid syntax is
-  described at https://github.com/grpc/grpc/blob/master/doc/naming.md.
+## Arrow Configuration
 
-## Advanced Configuration
+To use OTel-Arrow, a `grpc` configuration block must be listed.
 
-Several helper files are leveraged to provide additional capabilities automatically:
+The `arrow` configuration block contains the following settings:
 
-- [gRPC settings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configgrpc/README.md) including CORS
-- [TLS and mTLS settings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md)
-
-## Writing with HTTP/JSON
+- `disabled` (default: false) prevents serving OTel-Arrow requests, allowing only standard OTLP requests.
+- `disable_separate_signals` (default: false) when true, signal-specific (i.e., traces, metrics, logs) OTel-Arrow streams are not served.
+- `disable_mixed_signals` (default: false) when true, mixed-signal (i.e., traces, metrics, logs) OTel-Arrow streams are not served.
```

Comment:
I'm uncertain about the purpose of this flag. Providing a use case example might clarify its function.

Reply:
I removed all the disable features.
This makes the change larger because a lot of tests were ensuring that "grpc" was explicitly selected which is now not needed. I updated README to indicate that no configuration is required.
However, to reduce the size of the diff and keep merges simple, the three fields in `Protocols` are still pointers, and they can still be disabled explicitly in code by setting the fields to nil. It's no longer possible to disable gRPC or Arrow via configuration, only an explicit `field = nil` assignment will do now.